src/core/namespace.c

```diff
@@ -1576,13 +1588,23 @@ int setup_namespace(
                         n_mount_images,
                         tmp_dir, var_tmp_dir,
                         creds_path,
-                        log_namespace);
+                        log_namespace,
+                        setup_propagate,
+                        root_is_read_only);
 
         if (n_mounts > 0) {
                 m = mounts = new0(MountEntry, n_mounts);
                 if (!mounts)
                         return -ENOMEM;
 
+                /* We need at least /run to be writable, so add a tmpfs in case the user doesn't */
```

Comment:
This is the part I'm less sure about - always having /run as a tmpfs is a big change, even if restricted to cases where root is r/o.
Not sure how else to handle it though - perhaps using /tmp instead if /run is readonly?

Reply:
On the other hand, if a manual mount from the host is requested by the user, it will overwrite this, so perhaps it's not so bad. Thoughts?

Reply:
I'm not sure either. But I think this approach of making it "just work" is the least bad.